src/logger.js

```diff
@@ -211,65 +195,72 @@ class Logger extends Writable {
   }
 
   /**
-   * Override Writable _write method.
+   * Override {Writable} _write method.
    * Get the connection promise .then write the next log on the ringBuffer
-   * to Logentries connection when its available
+   * to the Insight Platform connection when its available
    */
   _write(ch, enc, cb) {
     this.drained = false;
-    this.connection.then(conn => {
+    this.connection.then((conn) => {
       const record = this.ringBuffer.read();
+
       if (record) {
         // we are checking the buffer state here just after conn.write()
         // to make sure the last event is sent to socket.
         if (this.ringBuffer.isEmpty()) {
           conn.write(record, () => {
             process.nextTick(() => {
               this.emit(bufferDrainEvent);
-              // this event is DEPRECATED - will be removed in next major release.
-              // new users should use 'buffer drain' event instead.
-              this.emit('connection drain');
             });
           });
         } else {
           conn.write(record);
         }
       } else {
+        //  This from my experience, means we have not written to the
+        //  ringBuffer, which means that reading will return `null`.
         this.debugLogger.log('This should not happen. Read from ringBuffer returned null.');
       }
       cb();
-    }).catch(err => {
+    }).catch((err) => {
       this.emit(errorEvent, err);
       this.debugLogger.log(`Error: ${err}`);
       cb();
     });
   }
 
-  setDefaultEncoding() { /* no. */
+  //  Here we want to overwrite the setDefaultEncoding method but eslint will complain
+  // if we don't use `this` in a class method (should be static in that scenario), so here we
+  // ignore the error
+  /* eslint class-methods-use-this: ["error", { "exceptMethods": ["setDefaultEncoding"] }] */
+  setDefaultEncoding() {/* no. */
```

Comment:
In case it complains about it being an empty function (which I think it should but the comment might be preventing that);
`setDefaultEncoding = () => void 0;`

Reply:
nah it's not complaining about being empty, described in the comment, it just complains that it's a class function and doesn't use `this`, therefore should be a static function